woocommerce/woocommerce-ios
# [Tablet Orders] Use recalculate button for selection in side-by-side mode

Part of: #11908


## Description


This PR moves our side-by-side order creation experience to use a `Recalculate` button, instead of the `immediate` sync approach. Users must tap `Recalculate` in side-by-side mode to sync the order changes before they can continue with creating the order.

It’s added as temporary mitigation for selection sync difficulties.

We’ll go back to immediate in the near future, so that code is left in for ease later.

Note that this PR also moves the responsibility for triggering sync to the EditableOrderViewModel, via the previously existing closures such as `onProductSelectionStateChanged` or `onAllSelectionsCleared`. This seems like less of a side-effect than having the Product Selector call `onMultipleSelectionCompleted` at appropriate times.

## Testing instructions


1. Enable the `sideBySideViewForOrderForm` feature flag
2. Launch the app on an iPad or large iPhone in landscape
3. Navigate to the `Orders` tab, and tap `+` to open the order form
4. Add products to the order
5. Observe that the button shows `Recalculate`
6. Tap `Recalculate` – observe that the order summary is updated, and the completion button becomes `Collect Payment`.

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/2472348/df7e3c2d-9a78-4844-a598-f18575edec83


---
